ols/user_data_collection/data_collector.py

```diff
@@ -0,0 +1,313 @@
+            "application/vnd.redhat.openshift.periodic+tar",
+        ),
+    }
+
+    if CP_OFFLINE_TOKEN:
+        logger.debug("using CP offline token to generate refresh token")
+        token = access_token_from_offline_token(CP_OFFLINE_TOKEN)
+        cluster_id = "unknown"
+    else:
+        logger.debug("using cluster pull secret to authenticate")
+        token = get_cloud_openshift_pull_secret()
+        cluster_id = get_cluster_id()
+
+    with requests.Session() as s:
+        s.headers = {
+            # TODO: OLS-486 User agent needs to be set as insights operator,
+            # otherwise JWT token is not accepted - we use just random hash
+            # "User-Agent": f"openshift-lightspeed/user-data-collection cluster/{cluster_id}",
+            "User-Agent": f"insights-operator/abcdefgijklmnopqrstuvwxyz01234567891011 cluster/{cluster_id}",  # noqa: E501
+            "Authorization": f"Bearer {token}",
+        }
+        if INGRESS_ENV == "stage":
+            # stage doesn't accept user-agent
+            del s.headers["User-Agent"]
+
+        response = s.post(
+            url=get_ingress_upload_url(),
+            files=payload,
+            timeout=INGRESS_TIMEOUT,
+        )
+
+    if response.status_code == 202:
+        request_id = response.json()["request_id"]
+        logger.info(f"data uploaded with request_id: '{request_id}'")
+    else:
+        raise requests.exceptions.HTTPError(
+            f"data upload failed with response: {response.json()}"
+        )
+
+    return response
+
+
+def delete_data(file_paths: list[pathlib.Path]) -> None:
+    """Delete files from the provided paths.
+
+    Args:
+        file_paths: List of paths to the files to be deleted.
+    """
+    for file_path in file_paths:
+        logger.debug(f"removing '{file_path}'")
+        file_path.unlink()
+        if file_path.exists():
+            logger.error(f"failed to remove '{file_path}'")
+
+
+def gather_ols_user_data(data_path: str) -> None:
+    """Gather OLS user data and upload it to the Ingress service."""
+    collected_files = collect_ols_data_from(data_path)
+    if collected_files:
+        logger.info(f"collected {len(collected_files)} files from '{data_path}'")
+        tarball = package_files_into_tarball(collected_files, path_to_strip=data_path)
+        upload_data_to_ingress(tarball)
+        delete_data(collected_files)
+        logger.info("uploaded data removed")
+        # TODO: OLS-473
+        # ensure_data_folder_is_not_bigger_than(OLS_USER_DATA_MAX_SIZE)
+    else:
+        logger.info(f"'{data_path}' contains no data, nothing to do...")
+
+
+if __name__ == "__main__":
+    # TODO: Temporary solution until OLS-500
+    if INGRESS_ENV == "stage" and not CP_OFFLINE_TOKEN:
+        try:
+            get_cloud_openshift_pull_secret()
+        except KeyError:
+            logger.error(
+                "Cluster pull secret does not contain the cloud.openshift.com auth"
+            )
+            sys.exit(1)
```

Comment:
noted elsewhere but to put it all in one review:  if you do this the sidecar will crashloop which is not what we want.

if we can't get the auth key from the pullsecret we should just log the error and retry on the next regular interval.

1) the key might show up later
2) the key might be removed later